src/java.base/share/classes/java/lang/ref/Reference.java

```diff
@@ -511,60 +541,78 @@ protected Object clone() throws CloneNotSupportedException {
     }
 
     /**
-     * Ensures that the object referenced by the given reference remains
-     * <a href="package-summary.html#reachability"><em>strongly reachable</em></a>,
-     * regardless of any prior actions of the program that might otherwise cause
-     * the object to become unreachable; thus, the referenced object is not
+     * Ensures that the given object remains
+     * <a href="package-summary.html#reachability"><em>strongly reachable</em></a>.
+     * This reachability is assured regardless of any optimizing transformations
+     * the virtual machine may perform that might otherwise allow the object to
+     * become unreachable (see JLS {@jls 12.6.1}). Thus, the given object is not
      * reclaimable by garbage collection at least until after the invocation of
-     * this method.  Invocation of this method does not itself initiate garbage
-     * collection or finalization.
+     * this method. References to the given object will not be cleared (or
+     * enqueued, if applicable) by the garbage collector until after invocation
+     * of this method.
+     * Invocation of this method does not itself initiate reference processing,
+     * garbage collection, or finalization.
      *
      * <p> This method establishes an ordering for <em>strong reachability</em>
      * with respect to garbage collection.  It controls relations that are
      * otherwise only implicit in a program -- the reachability conditions
-     * triggering garbage collection.  This method is designed for use in
-     * uncommon situations of premature finalization where using
-     * {@code synchronized} blocks or methods, or using other synchronization
-     * facilities are not possible or do not provide the desired control.  This
-     * method is applicable only when reclamation may have visible effects,
-     * which is possible for objects with finalizers (See Section {@jls 12.6}
-     * of <cite>The Java Language Specification</cite>) that
-     * are implemented in ways that rely on ordering control for
-     * correctness.
+     * triggering garbage collection.  This method is applicable only
+     * when reclamation may have visible effects,
+     * such as for objects that use finalizers or {@link Cleaner}, or code that
```

Comment:
This `@link` tag [does not produce a proper link in the Javadoc](https://docs.oracle.com/en/java/javase/23/docs/api/java.base/java/lang/ref/Reference.html#reachabilityFence(java.lang.Object)) because this file imports `jdk.internal.ref.Cleaner`.